### DISCUSSION TITLE
Load icon from a file?

### DISCUSSION THREAD

**Discussion by daladim** [hecrj/iced#1216] [Q&A] [answered]
Thanks for iced.

I'm using iced to make a native (Windows and Linux if that matters) app, and I'd like to give its window an icon.
I managed to add an icon using [Icon](https://docs.rs/iced/latest/iced/window/icon/struct.Icon.html) and passing it to my app's [`settings`](https://docs.rs/iced/latest/iced/window/struct.Settings.html#structfield.icon).

Is there a way (I haven't found in the doc, but I may not have been looking at the right places) to load an icon from a file, instead of giving a pre-computed array of pixels?

Especially, I'd like to load an SVG icon. Since you already provide [SVG widgets](https://docs.rs/iced/latest/iced/widget/svg/struct.Svg.html). I haven't read its source code yet, but maybe this could be somehow reused so that we could provide an `Icon::from_file()`, or at least `Icon::from_svg()`.
What do you think? Would that make sense?
Is there an easy way to rasterize an SVG into a pixel array already, and would it be better suited to build the array first, then give it to `Icon::from_rgba`?

I could help and make the MR myself, once you tell me what solution sounds the most sensible to you.
Thanks

**Comment by daladim** [comment]
How does [`svg::Handle::data()`](https://docs.rs/iced/latest/iced/widget/svg/struct.Handle.html#method.data) relates to an RGBA array?

**Comment by 13r0ck** [comment, marked as answer]
This is how I did it in my project

1. [first create the icon data](https://github.com/13r0ck/bathtub/blob/main/src/main.rs#L39)
2. [pass that into the iced window](https://github.com/13r0ck/bathtub/blob/main/src/main.rs#L47)

**Reply by daladim** [reply]
Thanks.
That's more-or-less what I'm doing in https://github.com/iced-rs/iced/pull/1174
